brayns/io/ProteinLoader.cpp

```diff
@@ -434,27 +434,36 @@ bool ProteinLoader::importPDBFile(const std::string& filename,
                     ++i;
                 }
 
-                const auto materialId =
-                    colorScheme == ColorScheme::protein_by_id
-                        ? proteinIndex % scene.getMaterials().size()
-                        : atom.materialId;
                 // Convert position from nanometers
-                const auto center = position + 0.01f * atom.position;
+                const auto center =
+                    transformation.getTranslation() + 0.01f * atom.position;
+
                 // Convert radius from angstrom
                 const auto radius = 0.0001f * atom.radius *
                                     _geometryParameters.getRadiusMultiplier();
-                scene.addSphere(materialId, {center, radius});
+
+                const auto materialId =
+                    colorScheme == ColorScheme::protein_by_id ? index
+                                                              : atom.materialId;
+                spheres[materialId].push_back({center, radius});
             }
         }
         file.close();
-    }
-
-    return true;
-}
 
-Vector3f ProteinLoader::getMaterialKd(const size_t index)
-{
-    return Vector3f(colorMap[index].R / 255.f, colorMap[index].G / 255.f,
-                    colorMap[index].B / 255.f);
+        // Add materials and spheres
+        size_t i = 0;
+        for (const auto& spheresPerMaterial : spheres)
+        {
+            const auto materialId = spheresPerMaterial.first;
+            auto material =
+                model->createMaterial(materialId, colorMap[i].symbol);
+            material->setDiffuseColor({colorMap[materialId].R / 255.f,
+                                       colorMap[materialId].G / 255.f,
+                                       colorMap[materialId].B / 255.f});
+            for (const auto& sphere : spheresPerMaterial.second)
```

Comment:
Why do you use colorMap[i] and colorMap[materialId] for the same material?

Reply:
Fixed!